### PR TITLE
Turned on strictNullChecks, strictPropertyInitialization for fast-web…

### PR DESCRIPTION
**Pull request by NiteshSethi09**
…-utilities in tsconfig

# Pull Request

## 📖 Description

Made changes in tsconfig.json and removed all the eslint errors. Made relevant changes and passed all the test cases.

## 👩‍💻 Reviewer Notes

I made some parameters value as non-null value and made all the tests cleared. Some methods like ```substr()``` were showing deprecated, changed them to the updated and working ones. Need to change the other ones also as there could be more. 

### General

- [ ] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [x] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

File: packages/utilities/fast-web-utilities/src/strings.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ export function startsWith(
         return false;
     }
 
-    return stringToSearch.substr(position, searchFor.length) === searchFor;
+    return stringToSearch.substring(position, searchFor.length) === searchFor;
```
Comment (chrisdholt) [inline packages/utilities/fast-web-utilities/src/strings.ts:44]: I don't think a change like this is trivial - technically this change could constitute a breaking change. While `substr() ` and `substring()` are similar, they are different and could result in differences in what's being returned: https://stackoverflow.com/a/3745518/4782111

**Review by chrisdholt — CHANGES_REQUESTED** [review]
Holding on this given the change to the the helper method for starts with. It think the included change is reasonably significant to introduce regressions without a breaking change.

File: packages/utilities/fast-web-utilities/src/strings.ts (hunk 1 above)
Comment (NiteshSethi09) [reply packages/utilities/fast-web-utilities/src/strings.ts:44]: Thanks for pointing out me @chrisdholt. I thought I've replaced a deprecated method with a working one. Though now I have a look at what I have done. Will try to look forward from next time onwards before changing anything.

**Comment by NiteshSethi09** [comment]
@nicholasrice May we look here also for a resolution and work together to get it resolved?

**Pull request closed without merging**